Lib/ctypes/__init__.py

```diff
@@ -550,10 +552,9 @@ def LoadLibrary(self, name):
 
 if _os.name == "nt":
     pythonapi = PyDLL("python dll", None, _sys.dllhandle)
-elif _sys.platform == "android":
-    pythonapi = PyDLL("libpython%d.%d.so" % _sys.version_info[:2])
-elif _sys.platform == "cygwin":
-    pythonapi = PyDLL("libpython%d.%d.dll" % _sys.version_info[:2])
+elif _sys.platform in ["android", "cygwin"]:
+    # These are Unix-like platforms which use a dynamically-linked libpython.
+    pythonapi = PyDLL(_sysconfig.get_config_var("LDLIBRARY"))
```

Comment:
fyi, on cygwin the DLL is `DLLLIBRARY` while `LDLIBRARY` is the import library, which wont work, see https://github.com/python/cpython/blob/a882ae198ade5008d42b8474a37409bf1502cf74/configure.ac#L1407-L1414